packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -1132,6 +1137,91 @@ function commitLayoutEffectOnFiber(
   }
 }
 
+function abortParentMarkerTransitions(
+  deletedFiber: Fiber,
+  nearestMountedAncestor: Fiber,
+  abort: TransitionAbort,
+) {
+  let fiber = deletedFiber;
+  let isInDeletedTree = true;
+  while (fiber !== null) {
+    const instance = deletedFiber.stateNode;
+    switch (fiber.tag) {
+      case TracingMarkerComponent:
+        const transitions = instance.transitions;
+
+        const markerInstance = fiber.stateNode;
+        const markerTransitions = markerInstance.transitions;
+        const abortMarker = Array.from(transitions).some(transition =>
+          markerTransitions.has(transition),
+        );
+
+        if (abortMarker) {
```

Comment:
If you do a normal for loop, you can put all this code inside the loop and then `break` out of it once you find a match. We tend to prefer that when possible instead of the functional array methods.

Reply:
We can't because transitions is a set